lib/src/rules/require_trailing_commas.dart

```diff
@@ -0,0 +1,168 @@
+// Copyright (c) 2021, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/token.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+
+import '../analyzer.dart';
+
+const _desc = r'Use trailing commas for all function calls and declarations.';
+
+const _details = r'''
+
+**DO** use trailing commas for all function calls and declarations unless the
+function call or definition, from the start of the function name up to the
+closing parenthesis, fits in a single line.
+
+**GOOD:**
+```dart
+void run() {
+  method(
+    'does not fit on one line',
+    'test test test test test test test test test test test',
+  );
+}
+```
+
+**BAD:**
+```dart
+void run() {
+  method('does not fit on one line',
+      'test test test test test test test test test test test');
+}
+```
+
+**Exception:** If the final parameter/argument is positional (vs named) and is
+either a function literal implemented using curly braces, a literal map, a
+literal set or a literal array. This exception only applies if the final
+parameter does not fit entirely on one line.
+
+**Note:** This lint rule assumes `dartfmt` has been run over the code.
+
+''';
+
+class RequireTrailingCommas extends LintRule implements NodeLintRule {
+  RequireTrailingCommas()
+      : super(
+          name: 'require_trailing_commas',
+          description: _desc,
+          details: _details,
+          group: Group.style,
+        );
+
+  @override
+  void registerNodeProcessors(
+    NodeLintRegistry registry,
+    LinterContext context,
+  ) {
+    var visitor = _Visitor(this);
+    registry
+      ..addCompilationUnit(this, visitor)
+      ..addArgumentList(this, visitor)
+      ..addFormalParameterList(this, visitor)
+      ..addAssertStatement(this, visitor)
+      ..addAssertInitializer(this, visitor);
+  }
+}
+
+class _Visitor extends SimpleAstVisitor<void> {
+  static const _trailingCommaCode = LintCode(
+    'require_trailing_commas',
+    'Missing a required trailing comma.',
+  );
```

Comment:
I don't think you need to create your own lint code instance and can just call `rule.reportLintForToken(token)` (which will pickup the default lint code implementation returned by the `lintCode` getter which should be the same as what you're creating here).

Reply:
(Places where you *do* want your own lint codes are when you want to define more than one to capture additional context. See for example: https://github.com/dart-lang/linter/blob/983cbf8b2f8a5fd27658e20f0871ac8e8c0276b9/lib/src/rules/always_declare_return_types.dart#L67-L77)

Reply:
FWIW, I think that this is a much better message than the default (which would be "Use trailing commas for all function calls and declarations.").

Reply:
Agree with @bwilkerson here.

Reply:
Whoops!  I didn't see that. 😬   I totally agree. Far better.  Sorry for the noise!